runelite-client/src/main/java/net/runelite/client/plugins/timers/TimersPlugin.java

```diff
@@ -516,12 +517,22 @@ public void createGameTimer(GameTimer timer)
 		removeGameTimer(timer);
 
 		TimerTimer t = new TimerTimer(timer, this);
-		t.setTooltip(timer.getDescription());
+		t.setTooltip(getTooltipDescription(timer));
```

Comment:
I did not meant moving the logic to separate function. I meant adding subclass of timer and make it like CannonTimer or something and sending the cannonWorld there.